c/blake3_impl.h

```diff
@@ -43,10 +43,20 @@ static const uint8_t MSG_SCHEDULE[7][16] = {
     {11, 15, 5, 0, 1, 9, 8, 6, 14, 10, 2, 12, 3, 4, 7, 13},
 };
 
+INLINE uint32_t counter_low(uint64_t counter) { return (uint32_t)counter; }
+
+INLINE uint32_t counter_high(uint64_t counter) {
+  return (uint32_t)(counter >> 32);
+}
+
 // Count the number of 1 bits.
 INLINE uint8_t popcnt(uint64_t x) {
 #if __POPCNT__
+#ifdef __x86_64__
```

Comment:
Do we need to check an extra variable for MSVC support here? In `blake3_dispatch.c` we also check `_M_X64`. Then again, maybe we have the same problem with the `__POPCNT__` variable.

Makes me wonder if this optimization is even worth it. The portable version is almost certainly not a hot spot.

Reply:
I did a quick test using a [msvc online compiler](https://rextester.com/l/cpp_online_compiler_visual) that I found and it looks like `__POPCNT__` isn't defined (or I just don't know the correct compiler argument to use).

Reply:
MSVC has no way to detect popcnt at compile-time.